### PR TITLE
Working on Category API 

### PR DESCRIPTION
**Pull request by sonikavirani**
In Category API add the category ,findAll,findById,update,delete and also findByTitle but issue occurred in findByTitle

### CODE REVIEW

File: src/action/categoryAction.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const CategoryModel = require('../model/category');
+
+module.exports = {
+  insertCategory: function (req, res) {
+    const {title} = req.body;
+    const categoryData = {
+      title,
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: use single line for create object if you have only single value

File: src/action/categoryAction.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  /*findById: function (req, res) {
+    const {id} = req.params;
+    CategoryModel.findById(id).then((data) => {
+      res.json({status: true, data: data})
+    })
+  },*/
+
+  getCategory: function (req, res) {
+    const {title} = req.params;
+    console.log("rqe...", req.params)
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: Remove console before commit from all files

File: src/action/categoryAction.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  },
+
+  /*findById: function (req, res) {
+    const {id} = req.params;
+    CategoryModel.findById(id).then((data) => {
+      res.json({status: true, data: data})
+    })
+  },*/
+
+  getCategory: function (req, res) {
```
Comment (suresh-jarvis) [inline src/action/categoryAction.js]: There is no route to find category by title.
Use id in route

File: src/action/usersAction.js [hunk 4]
```diff
@@ -1,31 +1,31 @@
...
       res.json({status: true, data: data});
     })
   },
 
   delete: function (req, res) {
     const {id} = req.params;
     console.log(req.params);
-    empModel.delete(id).then(() => {
+    userModel.delete(id).then(() => {
       res.json({status: true, message: 'Deleted SuccessFully!!!!!...'})
```
Comment (suresh-jarvis) [inline src/action/usersAction.js]: Do proper message No caps after first word and No exrta sign after message.
`res.json({status: true, message: 'User deleted successfully.'})`

File: src/action/usersAction.js [hunk 5]
```diff
@@ -55,7 +55,7 @@ module.exports = {
           isVerified: true,
           email
         };
-        empModel.insert(userData).then((data) => {
+        userModel.insert(userData).then((data) => {
           if (Object.keys(data).length > 0) {
             res.json({status: true, data: data, message: 'SignUp SuccessFully!!!!'})
```
Comment (suresh-jarvis) [inline src/action/usersAction.js]: Proper message string please

File: src/app.js [hunk 6]
```diff
@@ -15,8 +15,9 @@ router.use(function (req, res, next) {
   next();
 });
 
-app.use('/employee', require('./router/employeeRoute'));
+app.use('/users', require('./router/usersRoute'));
 app.use('/product', require('./router/productRoute'));
+app.use('/category', require('./router/categoryRoute'));
```
Comment (suresh-jarvis) [inline src/app.js]: If you are using `s` in module routes then use in whole app
Like : /users
/products
/categories

File: src/model/users.js [hunk 7]
```diff
@@ -1,58 +1,60 @@
...
+      users.findById(id, function (err, data) {
         resolve((err) ? {} : (data == null) ? {} : data);
       })
     })
   },
 
-  findByCredential: function (email,password) {
+  findByCredential: function (email, password) {
     return new Promise((resolve) => {
       console.log('email ------------', email, password);
```
Comment (suresh-jarvis) [inline src/model/users.js]: remove console

File: src/router/categoryRoute.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+const express = require('express');
+const router = express.Router();
+const categoryAction = require('../action/categoryAction');
+
+router.post('/', categoryAction.insertCategory);
+router.get('/', categoryAction.findAll);
+/*router.get('/:id', categoryAction.findById);*/
+router.get('/:title', categoryAction.getCategory);
```
Comment (suresh-jarvis) [inline src/router/categoryRoute.js]: Replace with Id

File: src/action/categoryAction.js (hunk 1 above)
Comment (sonikavirani) [reply src/action/categoryAction.js]: Done this changes

File: src/action/categoryAction.js (hunk 2 above)
Comment (sonikavirani) [reply src/action/categoryAction.js]: Okk..I will do it.

File: src/action/usersAction.js (hunk 4 above)
Comment (sonikavirani) [reply src/action/usersAction.js]: Okk..
I Will follow as per your comments.

File: src/action/usersAction.js (hunk 5 above)
Comment (sonikavirani) [reply src/action/usersAction.js]: Okk..
I Will Paste the proper message.

File: src/model/users.js (hunk 7 above)
Comment (sonikavirani) [reply src/model/users.js]: I will Remove the console.

File: src/app.js (hunk 6 above)
Comment (sonikavirani) [reply src/app.js]: Done this changes.

File: src/router/categoryRoute.js (hunk 8 above)
Comment (sonikavirani) [reply src/router/categoryRoute.js]: Okk..Replace title with id.

**Pull request closed without merging**